libs/external-data/src/readability/readability.service.ts

```diff
@@ -0,0 +1,220 @@
+      const lang = langRegex.exec(fullUrl)[1];
+
+      const preText = mainContent
+        .replace('..', '.')
+        .replace('.', '. ')
+        .slice(0, mainContent.indexOf('defPreFooter'))
+        .replace(/\n/g, '')
+        .replace(/\t/g, '')
+        .replace(/\r/g, '');
+
+      // add periods after bullet points and headings so that flesh kincaid considers them as sentences
+      const html1 = html.replace(/<\/(li|h[1-6]|p)>/g, '.$&');
+
+      const revText = load(html1);
+      revText('table, div.pagedetails, #chat-bottom-bar').remove();
+      const textContent = revText('main').text();
+
+      const postText = textContent
+        .replace(/\.\./g, '. ')
+        .replace(/(\.)\.\s/g, '$1')
+        .replace(/\s+\.\s|\s+\./g, '. ')
+        .replace(/([.]+)(?=$|\s)/g, '. ')
+        .replace(/\n|\t|\r/g, '')
+        .replace(/([.:;!?])\s*\./g, '$1 ')
+        .replace(/\s+/g, ' ');
+
+      const word = readability.lexiconCount(postText);
+      const sentence = this.sentenceCount(postText);
+      let syllables: number;
+      let original_fk: number;
+      let final_fk: number;
+
+      if (lang === 'fr') {
+        syllables = this.syllableCountFr(postText);
+        original_fk = this.fleschKincaidGradeFr(preText);
+        final_fk = this.fleschKincaidGradeFr(postText);
+      } else {
+        syllables = readability.syllableCount(postText);
+        original_fk = readability.fleschKincaidGrade(preText);
+        final_fk = readability.fleschKincaidGrade(postText);
+      }
+
+      // HTML string to be parsed
+      const mainHtml = load(main.html());
+
+      // get all headings and calculate how many words on average between headings
+      const headings = mainHtml('h1, h2, h3, h4, h5, h6');
+      const len_headings = headings.length;
+      const hratio = word / len_headings;
+
+      // get all paragraphs and all bulleted list, and calculate how many words per paragraph on average
+      const paragraphs = mainHtml('p, ul');
+      const len_par = paragraphs.length;
+      const pratio = word / len_par;
+
+      let fkpoints: number;
+      if (final_fk <= 6) fkpoints = 60;
+      else if (final_fk >= 18) fkpoints = 0;
+      else fkpoints = this.toFixedNumber(60 - (final_fk - 6) * 5, 2);
+
+      // calculate points for number of words between headings
+      let hpoints: number;
+      if (hratio <= 40) hpoints = 20;
+      else if (hratio >= 200) hpoints = 0;
+      else hpoints = this.toFixedNumber(20 - (hratio - 40) * 0.125, 2);
+
+      // calculate points for number of words per paragraph
+      let ppoints: number;
+      if (pratio <= 30) ppoints = 20;
+      else if (pratio >= 80) ppoints = 0;
+      else ppoints = this.toFixedNumber(20 - (pratio - 30) * 0.4, 2);
+
+      // add all points
+      const total_score = fkpoints + hpoints + ppoints;
+
+      const words = await this.removeStopwords(postText, lang);
+      const wordCount = await this.wordCount(words);
+
+      const dataWord =
```

Comment:
```suggestion
      const wordCounts =
```
Because I had no idea what `dataWord` meant